src/routes/ValidRoute.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import { useSelector, shallowEqual } from 'react-redux';
+import { useHistory } from 'react-router';
+
+import Unauthorized from '../Features/Unauthorized';
+import { isProtectedRoute } from './Permission';
+
+const ValidRoute = ({ route, authorizedRole }) => {
+  const { isLoggedIn, role } = useSelector((state) => {
+    const {
+      authReducer: { isLoggedIn, role },
+    } = state;
+    return {
+      role,
+      isLoggedIn,
+    };
+  }, shallowEqual);
+
+  const history = useHistory();
+  const { permissions, component: Component } = route;
+
+  if (isLoggedIn && permissions === isProtectedRoute && role === authorizedRole) {
+    return <Component />;
+  }
+
+  if (!isLoggedIn && window.location.pathname === '/login') history.push('/login');
```

Comment:
Why are u checking the window.location.pathname ??

Reply:
What I understand is .. It does check that user is on login page and again push the user to the login .. Am I right ?? Or there was a reason to do that ??

Reply:
outdated code